### DISCUSSION TITLE
Lucid postgres not setting up properly

### DISCUSSION THREAD

**Discussion by carsonwright** [adonisjs/core#1777] [Help]
When I follow these directions

```
yarn add @adonisjs/lucid@alpha
```
```
node ace invoke @adonisjs/lucid
```
I have postgres installed but it does not allow me to use
```
node ace make:model users
```
or
```
node ace make:migration users
```

I am currently using the api version as I don't need a front end. Any help would be appreciated.
These are my currently available commands:
```
Available commands
  build                 Compile typescript code to Javascript. Optionally watch for file changes
  invoke                Invoke post install instructions on a given AdonisJS package
  serve                 Compile typescript code to Javascript and start the HTTP server

dump
  dump:rcfile           Dump contents of .adonisrc.json file along with defaults

generate
  generate:key          Generate a new APP_KEY secret
  generate:manifest     Generate manifest file to execute ace commands

list
  list:routes           List application routes

make
  make:command          Make a new ace command
  make:controller       Make a new HTTP controller
  make:exception        Make a new exception handle
  make:listener         Make a new event listener class
  make:middleware       Make a new middleware
  make:prldfile         Make a new preloaded file. Preloaded files are loaded automatically on boot
  make:provider         Make a new IoC container provider
  make:validator        Make a new validator
  make:view             Make a new view template

Global Flags
  -h, --help boolean      
  -v, --version boolean  
```

**Comment by carsonwright** [comment]
I have figured out the issue, people referenced restarting the adonis server. I was not even running it at the time and didn't want to until I was able to run my desired command.

This I assume is because its in alpha but you actually have to run the server for it to finish setting up lucid.

**Reply by RomainLanz** [reply]
You are almost correct.

Since AdonisJS is using TypeScript and the runtime is using the compiled code, you must compile the code once to be able to see your change.

We have a section on the documentation explaining this.

> Since all of the AdonisJS runtime behavior relies on the compiled JavaScript code. It is advised to always have a terminal session compiling the TypeScript source.

**Reply by thetutlage** [reply]
Yo, as @RomainLanz said. However, we are getting rid of this compile time all together in development and hence it will remove all this unnecessary confusion.

**Reply by carsonwright** [reply]
That would be amazing @thetutlage I was thinking if at least ace somehow reran compilation or something that would be amazing.

**Reply by carsonwright** [reply]
@RomainLanz yeah I noticed it needed to be compiled for everything ha. I actually changed config stuff and was confused when the migrations didn't work. But I get it now. I'm kind of jumping in headfirst and have definitely not read all the warning signs etc. I also have been interested in this repo's health. Didn't want to buy into a framework that the maintainers weren't interested in.

Thank you all so much, so far this has been amazing!
